nengo_loihi/decode_neurons.py

```diff
@@ -178,16 +185,27 @@ def get_ensemble(self, dim):
                 "Input neurons with more than one neuron per dimension"
             )
 
+        for param in [
+            "n_neurons",
+            "dimensions",
+            "neuron_type",
+            "encoders",
+            "gain",
+            "bias",
```

Comment:
Is this checking necessary? As far as I can tell this is only being used internally, and only with the `add_to_container` kwarg? If a developer did specify one of these then it should throw an error further down when the ensemble is created. It's also missing some like `max_rates` and `intercepts`.

Reply:
The intention behind this commit is to make `DecodeNeurons` "user accessible", so users could pass any ensemble arguments they want. The reason to be defensive here is because the error further down will be confusing, it'll be a `TypeError` that'll make it seem like they passed something in twice, but in actuality we pass it in once and they pass it in once, so it just won't work.

We don't pass in everything though, including `max_rates` and `intercepts`, though we should change this in some way because we're now tripping the warning for setting gain, bias, and one of max_rates/intercepts, which we shouldn't.

Reply:
Decision from the meeting is to remove the `**kwargs` here and just allow passing in `add_to_container`. This makes things simpler for us, and is if anything more convenient for the user, as they can just modify parameters once they have the ensemble if they like.

Reply:
Done in fe055cee